core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -136,7 +137,9 @@ private[spark] class TaskSchedulerImpl(
   // IDs of the tasks running on each executor
   private val executorIdToRunningTaskIds = new HashMap[String, HashSet[Long]]
 
-  private val executorsPendingDecommission = new HashMap[String, ExecutorDecommissionInfo]
+  val executorsPendingDecommission = new HashMap[String, ExecutorDecommissionInfo]
+  // map of second to list of executors to clear form the above map
+  val decommissioningExecutorsToGc = new util.TreeMap[Long, mutable.ArrayBuffer[String]]()
```

Comment:
Could we use https://guava.dev/releases/17.0/api/docs/com/google/common/cache/CacheBuilder.html or other similar class that handles the GC cleanup for us? We already use CacheBuilder some places inside of Spark which is why I suggest it. (side note: I hadn't seen Java's TreeMap before, and it's a really neat use here).

Reply:
Sure. Any structure that lets me GC by time will do. I just wanted something lightweight and custom to this use case. 

I expect the treemap to contain no more than 60 seconds worth of entries since things are keyed by the second, and they are also cleaned up on every check. The check happens on every executor loss and fetch failures. But yeah it is possible that if there are no failures then the entries could just sit there :-P. 

I will change it to Cache. good idea.